lxd/container.go

```diff
@@ -836,7 +836,12 @@ func instanceCreateInternal(s *state.State, args db.InstanceArgs) (instance.Inst
 	}
 
 	if args.Profiles == nil {
-		args.Profiles = []string{"default"}
+		profiles, err := s.Cluster.ImageProfilesGet(args.BaseImage, args.Project)
+		if err != nil {
+			return nil, err
+		}
+		args.Profiles = profiles
+		logger.Infof("Got profiles: %v", args.Profiles)
```

Comment:
Logic looks good so with that in place, you shouldn't need that `lxc/init.go` change.